BatterySensor.h

```diff
@@ -68,20 +69,28 @@ class InternalVCC {
 
   void start () {
 #ifdef ARDUINO_ARCH_AVR
+    // TODO: dummy read - is this really necessary?
     // Read 1.1V reference against AVcc
     // set the reference to Vcc and the measurement to the internal 1.1V reference
     ADMUX &= ~(ADMUX_REFMASK | ADMUX_ADCMASK);
     ADMUX |= ADMUX_REF_AVCC;      // select AVCC as reference
     ADMUX |= ADMUX_ADC_VBG;       // measure bandgap reference voltage
+    ADCSRA |= (1 << ADSC);         // start conversion
+    while (ADCSRA & (1 << ADSC)) ; // wait to finish
+    (void) ADC;                    // dummy read
 #endif
-  }
-
+}
   uint16_t finish () {
     uint16_t vcc=0;
 #ifdef ARDUINO_ARCH_AVR
+    // Read 1.1V reference against AVcc
+    // set the reference to Vcc and the measurement to the internal 1.1V reference
+    ADMUX &= ~(ADMUX_REFMASK | ADMUX_ADCMASK);
+    ADMUX |= ADMUX_REF_AVCC;      // select AVCC as reference
+    ADMUX |= ADMUX_ADC_VBG;       // measure bandgap reference voltage
```

Comment:
_delay_us(250);

Laut https://jeelabs.org/2012/05/04/measuring-vcc-via-the-bandgap/  sollte man 250 µs warten, um eine stabile Spannung zu messen.
Im alten Code ist das falsch - da wird 350ms gewartet. Deshlab wollrw ch das ja auch aufteilen. aber 250µs sollte kein Problem sein.

Reply:
Ja, eine kurze Wartezeit mag ganz hilfreich sein.